### DISCUSSION TITLE
Tutto da zero!

### DISCUSSION THREAD

**Discussion by 19melos87** [indomus/forum#2614] [HUB: Home Assistant]
Ciao a tutti, apro questo topic perchè ho deciso di ricominciare da capo con home assistant e tutto il resto che ho a casa perchè tra ristrutturazione e lavori ho fatto molte cose in maniera frammentaria e non ci si capisce più niente! Da questo momento vorrei usare il topic per rifare tutto da capo con i vostri consigli e il vostro aiuto!

Sono curioso di vedere cosa mi suggerirete e cosa mi consiglierete come aggiunta!

Intanto vi elenco cosa ho in casa:

Come server su cui far girare HA ho questo:
- NiPoGi Mini PC W-11 Pro, Mini Computer 8GB DDR4 + 256GB M.2 SSD, AK1Plus 12th Intel Alder Lake-Ν95 (fino a 3.4GHz, 4C/4T), Gigabit Ethernet, 4K UHD, Dual Wi-Fi, BT 4.2

Inserito in un mini rack da 6 unità insieme ad uno switch 24 porte, un fritzbox 7530 e un nvr per le 4 telecamere che è questo:
- Reolink 4K Kit Videosorveglianza PoE, 4X 8MP + 8CH 2TB NVR per Registrazione 24/7, RLK8-812B4-A

Ho tutte le luci di casa collegate con degli shelly plus 1 e 2pm.
Ho uno shelly 3em collegato al quadro principale

Ho un chiller Olimpia Splendid Sherpa S2 che vorrei integrare

E poi come eletttrodomestici ho un frigo LG con wifi, forno e lavastoviglie Candy entrambe con wifi, ascugatrice hoover con wifi e lavatrice hoover smart ma senxza wifi (solo nfc).

Per finire come asssistente vocale utilizzo Alexa e ho anche installato un ring Intercom che rende smart il mio citofono urmet.

Intanto come inzio mi consigliate di mettere debian 12 direttamente sul mini pc oppure andare di proxmox e macchina virtuale?

**Comment by indomus** [comment]
> Intanto come inzio mi consigliate di mettere debian 12 direttamente sul mini pc oppure andare di proxmox e macchina virtuale?

assolutamente Linux Debian 12. Evita come la peste proxmox, non ne hai bisogno.

**Reply by mtorazzi** [reply]
perché addirittura "evita come la peste"?

un N95 con 8GB di RAM è assolutamente sovradimensionato per HomeAssistant, un proxmox permette di utilizzare lo stesso ferro in modo molto più efficiente, dividendo le risorse su varie macchine virtuali che si occupano di compiti diversi mantenendo la corretta segregazione, assegnando ad esempio risorse a un container con pihole, a una VM con jellyfin, a un server VPN o altro, senza investire in altro hardware...

**Comment by 19melos87** [comment]
Perfetto! Quindi metto Debian12 e vado si HA Core e metto tutto quello che di base trovo nelle guide indomus. Inotlre sto facendo il disegno della casa con sweet home 3d perchè mi piacerebbe avere la plancia come quelle che ho visto in qualche post. Poi cosa mi consigliate?

**Reply by indomus** [reply]
beh non c'è molto da consigliare, ha tutta roba facilmente integrabile e di qualità. dovresti andare come l'olio

**Reply by 19melos87** [reply]
Ok, intanto parto poi man mano vedrò se ho bisogno di una mano! Grazie

**Comment by 19melos87** [comment]
Ragazzi, dopo aver installato tutto ho quest'errore sul bluetooth. Come posso risolvere?
![image](https://github.com/indomus/forum/assets/12113379/18f30a09-c171-4abd-b6fe-8f23f8e90250)

**Reply by snakuzzo** [reply]
C'è scritto nel messaggio in grigio cosa devi fare

**Reply by 19melos87** [reply]
Si, ho letto l'errore. Ma cercando sembrerebbe che l'utente che esegue il container non abbia l'autorizzazione. Possibile sia questo il problema?

**Reply by snakuzzo** [reply]
Spiegati meglio...e fai vedere il compose

**Reply by 19melos87** [reply]
Ho letto cercando su google che se sul minipc il bluetooth funziona ed è visibile allora magari è un problema di permessi del docker che non sta prendendo la periferica e quindi qualcuno ha risolto aggiungendo la riga evidenziata con ** nel compose

```
  GNU nano 7.2                   docker-compose.yaml
version: '3.9'
services:
  home-assistant:
    container_name: home-assistant
    image: ghcr.io/home-assistant/home-assistant:stable
    volumes:
    - /home/homeassistant/.homeassistant:/config
    **- /run/dbus:/run/dbus:ro**
    environment:
    - TZ=Europe/Rome
    network_mode: host
    restart: always
```

**Reply by snakuzzo** [reply]
Non so dove tu abbia letto ciò, ma così com'è scritto non vuol dire niente...non ha alcun senso ed è sintatticamente errato.
Se provi anche solo a fare `up` ti darà

```
did not find expected alphabetic or numeric character
```

forse quelli che hanno scritto questa roba volevano semplicemente commentare quella riga con il cancelletto `#`, che comunque non è che risolve il problema...semplicemente non mappa l'adapter bluetooth dell'host con il container di Core.

Nei volumi va semplicemente aggiunto

```
- /run/dbus:/run/dbus:ro
```

e poi va ricreato il container. A quel punto verrà rilevato il bluetooth e verrà automaticamente rilevato l'adapter all'avvio di Core.

Ho appena ritestato anche io e funziona tutto regolarmente. Se non ti funziona, avrai fatto qualche altro errore

**Reply by 19melos87** [reply]
Forse mi sono spiegato male io, gli ** li ho aggiunti io per evidenziare la riga che ho trovato da aggiungere e che non c'era nel mio yaml. 
Adesso, dopo aver rilanciato il comando _docker compose up -d_  non c'è più traccia dell'errore.

Grazie!

**Comment by 19melos87** [comment]
Ho configurato  questi dispositivi:
![image](https://github.com/indomus/forum/assets/12113379/d5a206d1-2b84-4e52-ab35-d416020768e7)

Adesso ho questi ancora da configurare, a part Hyperion che l'ho installato su pc ma ancora devo collegare i vari componenti, gli altri dispositivi trovati li configuro tutti o vanno a crearmi entità doppioni che poi m fanno impazzire? 
![image](https://github.com/indomus/forum/assets/12113379/91deccb9-1793-472a-85d7-7cf82a2dd8e2)

**Reply by snakuzzo** [reply]
Ma guarda che non sei obbligato a configurare quella roba. Quella te la vede l'autodiscovery, ma puoi ignorare.
Il mio consiglio è...

Ti serve?
Se la risposta è no...ignora e buonanotte

**Reply by 19melos87** [reply]
Ok, ancora devo capire se l'integrazione del fritz mi possa essere utile. Se non erro posso usarla come tracker dei dispositivi che si connettono in wifi, poi cos'altro?

**Comment by 19melos87** [comment]
Avendo al momento sul mini pc con debian12 messo solo Home Assistant, cosa mi conosgiliate per poter avere un backup delle impostazioni e dei file principali così mentre faccio le varie posso sempre fare roll back?

**Reply by 19melos87** [reply]
Non volendo intasare la chat telegram con le mie richieste d'aiuto, provo a chiedere qui, anche se l'ultima mia domanda non ha mai avuto risposta.

Ho installato HA con docker su debian 12
Ho comprato un dominio su IONOS.it e su quest'ultimo ho scaricato il suo certificato SSL e funziona per accedere su HA con HTTPS.
Ora vorrei però mettere il sito sotto cloudflare come da voi consigliato e vorrei mettere il reverse proxy in modo da poter entrare in lan con http anche con l'indirizzo ip di HA e in https quando sono fuori.

Sto dicendo qualche caxxata?
Ho installato caddy e provato a configurarlo ma nei log vedo molti errori, sicuramente perchè manca un port forward che vorrei evitare usando la dns challenge o di IONOS o di Cloudflare, che dite?


EDIT:
Messo su cloudlare sembrava funzionare, ma dopo qualche minuto ha smesso. Se stacco i proxy da cloudflare torna a funzionare
